Makefile

```diff
@@ -41,6 +41,20 @@ skopeo: cmd/skopeo
 binary-local:
 	go build -ldflags "-X main.gitCommit=${GIT_COMMIT}" -o skopeo ./cmd/skopeo
 
+#   make debug
+#     Note: Uses the -N -l go compiler options to disable compiler optimizations
+#           and inlining. Using these build options allows you to subsequently
+#           use source debugging tools like delve.
+debug: debug-skopeo docs
+
+debug-skopeo: cmd/skopeo
+	docker build ${DOCKER_BUILD_ARGS} -f Dockerfile.build -t skopeobuildimage .
+	docker run --rm -v ${PWD}:/src/github.com/projectatomic/skopeo \
+		skopeobuildimage make debug-binary-local
+
+debug-binary-local:
+	go build -ldflags "-X main.gitCommit=${GIT_COMMIT}" -gcflags $(GOGCFLAGS) -o skopeo ./cmd/skopeo
+
```

Comment:
All of this duplication, and top-level targets, feels a bit unclean; would it be reasonably easy to have `make {binary,binary-local} DEBUG=1` or something like that, and use the DEBUG flag to decide whether to set the `-gcflags`?